route/table.go

```diff
@@ -295,13 +296,32 @@ func normalizeHost(host string, tls bool) string {
 
 // matchingHosts returns all keys (host name patterns) from the
 // routing table which match the normalized request hostname.
-func (t Table) matchingHosts(req *http.Request) (hosts []string) {
+func (t Table) matchingHosts(req *http.Request, cfg *config.Config) (hosts []string) {
```

Comment:
We are not looking to use Globs at all in the short term but we can spend some time testing this code and adding to a future PR to hedge against our requirements changing in the future.

Reply:
I understand that. I'd appreciate it if you could embed this code to the glob path after splitting the function and run some tests.

Reply:
Would it make more sense to add/delete the compiled glob at route add/route del?

Not sure what kind of performance impact that would have.  Especially in our kind of ENV where we have

Reply:
Since this is looping over the keys of t you could do this before the table is switched. However, my guess is that even with 1500 services your external hostnames are a bit more stable and you should have fewer.

Reply:
My gut feel says that the LRU cache is probably easier to handle here.